pipelines/common/src/main/java/com/google/fhir/analytics/view/ViewApplicator.java

```diff
@@ -407,11 +408,12 @@ public Builder addRow(FlatRow row) throws ViewApplicationException {
                   "New row size does not match schema: %d vs %d",
                   row.getElements().size(), columnInfos.size()));
         }
-        // Note the schema check is a little looser than what it should; in particular it is not
-        // sensitive to the order of columns in the new `row`. Or it does not care about some field
-        // mismatches, e.g., `description`.
-        for (RowElement e : row.getElements()) {
-          if (columnInfos.get(e.getName()) == null) {
+        // Checking that the new row has the same columns, in the same order. We could check extra
```

Comment:
@chandrashekar-s please note that I made the column checks a little more strict given that we now rely on those when writing to a DB. You raised this in a previous PR [here](https://github.com/google/fhir-data-pipes/pull/887#discussion_r1411487874) and now we can consider this done.

Reply:
Great! Thanks for the changes @bashir2.